File: ecmascript/bob/bob.js
//
// This is only a SKELETON file for the 'Bob' exercise. It's been provided as a
// convenience to get you started writing code faster.
//

class Bob {

	constructor() {
		this.QUESTION_ANSWER = 'Sure.';
		this.YELLING_ANSWER = 'Whoa, chill out!';
		this.BLANK_ANSWER = 'Fine. Be that way!';
		this.DEFAULT_ANSWER = "Whatever.";
	}

	isQuestion(message) {
		return message.slice(-1) == '?'
	}

	isYELLING(message) {
		return message == message.toUpperCase() && message !== message.toLowerCase()
	}

	hey(message) {
		//
		// YOUR CODE GOES HERE
		//
		message = message.trim()
		if (message === '') return this.BLANK_ANSWER
		if (this.isYELLING(message)) return this.YELLING_ANSWER
		if (this.isQuestion(message)) return this.QUESTION_ANSWER
		return this.DEFAULT_ANSWER


	}
}

export default Bob;

File: python/bob/bob.py
def hey(txt):

    QUESTION_ANSWER = 'Sure.'
    YELL_ANSWER = 'Whoa, chill out!'
    BLANK_ANSWER = 'Fine. Be that way!'
    DEFAULT_ANSWER = 'Whatever.'

    is_question = lambda x: x[-1] == '?'

    is_yelling = lambda x: x == x.upper() and x != x.lower()

    is_blank = lambda x: x == ''

    txt = txt.strip()
    
    if is_blank(txt):
        return BLANK_ANSWER
    elif is_yelling(txt):
        return YELL_ANSWER
    elif is_question(txt):
        return QUESTION_ANSWER
    else:
        return DEFAULT_ANSWER
    

    
